[WebGPU] Enable CTS in Debug, skipping tests which timeout

mwyrzykowski opened the pull request:
#### 6e3d7a52dc5f675749756551e68b203517ee930c
<pre>
[WebGPU] Enable CTS in Debug, skipping tests which timeout
<a href="https://bugs.webkit.org/show_bug.cgi?id=291530">https://bugs.webkit.org/show_bug.cgi?id=291530</a>
<a href="https://rdar.apple.com/149218906">rdar://149218906</a>

Reviewed by Tadeu Zagallo.

There is no release Sequoia EWS and these tests pass locally
but fail in nightly release Sequoia tests, which is kind of against
the goal of noticing regressions prior to commits landing.

Switch from Release -&gt; Slow and will mark any tests which continue
to fail in Debug Slow as Skip with this PR

* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/294307@main">https://commits.webkit.org/294307@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a6901c290a6298c6d6b75cb1edc8e32fce659b32

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101341 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21004 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11307 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106492 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51968 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21312 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29498 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77190 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34228 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104348 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16439 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91543 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57529 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16258 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9545 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51316 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86146 "Found 1 new API test failure: TestWebKitAPI.GPUProcess.ValidateWebAudioMediaProcessingAssertion (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9619 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108845 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28469 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20948 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86172 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28831 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87740 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85728 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21817 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30445 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8154 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/22583 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28399 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28211 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31531 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29769 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo commented on LayoutTests/platform/mac-wk2/TestExpectations:
```diff
@@ -1634,6 +1634,11 @@ http/tests/webgpu/webgpu/api/validation/image_copy/layout_related.html [ Skip ]
 http/tests/webgpu/webgpu/api/validation/render_pipeline/depth_stencil_state.html [ Skip ]
 http/tests/webgpu/webgpu/api/validation/render_pipeline/misc.html [ Skip ]
 http/tests/webgpu/webgpu/api/validation/resource_usages/texture/in_render_common.html [ Skip ]
+http/tests/webgpu/webgpu/api/validation/capability_checks/limits/maxSampledTexturesPerShaderStage.html [ Skip ]
```
Is it possible to skip these only in debug?

mwyrzykowski replied:
yes, that is a good idea

mwyrzykowski replied:
oh sorry @JonWBedard reports in https://bugs.webkit.org/show_bug.cgi?id=291527 that these are failing in Release too on the bots but not locally, perhaps we should Skip for now and then re-enable?

mwyrzykowski committed:
[WebGPU] Enable CTS in Debug, skipping tests which timeout
https://bugs.webkit.org/show_bug.cgi?id=291530
rdar://149218906

Reviewed by Tadeu Zagallo.

There is no release Sequoia EWS and these tests pass locally
but fail in nightly release Sequoia tests, which is kind of against
the goal of noticing regressions prior to commits landing.

Switch from Release -> Slow and will mark any tests which continue
to fail in Debug Slow as Skip with this PR

* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/294307@main
